pkg/config/identity/api_key.go

```diff
@@ -2,14 +2,94 @@ package identity
 
 import (
 	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/3scale-labs/authorino/pkg/common"
+	"github.com/3scale-labs/authorino/pkg/common/auth_credentials"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	apiKeySelector              = "api_key"
+	invalidApiKeyMsg            = "the API Key provided is invalid"
+	noApiKeysFoundMsg           = "no API Keys were found on the request"
+	authSuccessfulMsg           = "Successfully authenticated with the provided API key"
+	credentialsFetchingErrorMsg = "Something went wrong fetching the authorized credentials"
 )
 
+// APIKeyIdentityEvaluator interface represents the API Key Identity evaluator
+type APIKeyIdentityEvaluator interface {
```

Comment:
We have another one already defined at https://github.com/3scale-labs/authorino/blob/e0290e73d6963ce8b5b03e4a7e7edd0618913608/pkg/common/auth.go#L20, which makes it easier to be imported without circular refs. However, let's leave this one here for now and we can move it with the other interfaces of the kind in another PR.

Reply:
Hmm. I see we're not using the interface for anything just yet. So it's just for the good practice of communicating the contract.